commons-geometry-euclidean/src/main/java/org/apache/commons/geometry/euclidean/threed/hull/ConvexHull3D.java

```diff
@@ -0,0 +1,697 @@
+     * a path around the horizon.</li>
+     * <li>We construct a new cone of polygons from the edges of the horizon to the
+     * outside point. All visible facets are removed and the points in the outside
+     * sets of the visible facets are redistributed.</li>
+     * <li>We repeat step 3-5 until each outside set is empty.</li>
+     * </ol>
+     */
+    public static class Builder {
+
+        /** Set of possible candidates. */
+        private final PointSet<Vector3D> candidates;
+
+        /** Precision context used to compare floating point numbers. */
+        private final DoubleEquivalence precision;
+
+        /** Simplex for testing new points and starting the algorithm. */
+        private Simplex simplex;
+
+        /** The minX, maxX, minY, maxY, minZ, maxZ points. */
+        private final Vector3D[] box;
+
+        /**
+         * A map which contains all the vertices of the current hull as keys and the
+         * associated facets as values.
+         */
+        private Map<Vector3D, Set<Facet>> vertexToFacetMap;
+
+        /**
+         * Constructor for a builder with the given precision.
+         *
+         * @param precision the given precision.
+         */
+        public Builder(DoubleEquivalence precision) {
+            candidates = EuclideanCollections.pointSet3D(precision);
+            this.precision = precision;
+            vertexToFacetMap = EuclideanCollections.pointMap3D(precision);
+            box = new Vector3D[6];
+            simplex = new Simplex(Collections.emptySet());
+        }
+
+        /**
+         * Appends to the point to the set of possible candidates.
+         *
+         * @param point the given point.
+         * @return this instance.
+         */
+        public Builder append(Vector3D point) {
+            if (box[0] == null) {
+                box[0] = box[1] = box[2] = box[3] = box[4] = box[5] = point;
+                candidates.add(point);
+                return this;
+            }
+            boolean hasBeenModified = false;
+            if (box[0].getX() > point.getX()) {
+                box[0] = point;
+                hasBeenModified = true;
+            }
+            if (box[1].getX() < point.getX()) {
+                box[1] = point;
+                hasBeenModified = true;
+            }
+            if (box[2].getY() > point.getY()) {
+                box[2] = point;
+                hasBeenModified = true;
+            }
+            if (box[3].getY() < point.getY()) {
+                box[3] = point;
+                hasBeenModified = true;
+            }
+            if (box[4].getZ() > point.getZ()) {
+                box[4] = point;
+                hasBeenModified = true;
+            }
+            if (box[5].getZ() < point.getZ()) {
+                box[5] = point;
+                hasBeenModified = true;
+            }
+            candidates.add(point);
+            if (hasBeenModified) {
```

Comment:
I thought you would most likely go for this option only if you have most if not all points available beforehand and would most likely reconstruct the simplex anyway. But you are right it would be more consequential to add the same check here as well.

Reply:
To be honest i would keep this different behavior and and document it better. On my machine test were significantly lower with the same beahvior. With the same behavior the test suite takes 6.3s and without it is only 4.9s

<img width="161" alt="image" src="https://github.com/apache/commons-geometry/assets/75903169/3bbda548-739d-4815-aff5-290b60fa034e">

<img width="169" alt="image" src="https://github.com/apache/commons-geometry/assets/75903169/ceae4e3e-ee9c-45be-8030-421922b6933e">

Reply:
Maybe this is an indication to go for another test while adding points, which more closely resembles the original algorithm. We could collect a certain number of points and then perform the algorithm. After that, we could reset the counter, test the newly added point against the existing structure, and perform the algorithm again after the counter passes the threshold.

Reply:
I added a similar check as in the append(Vector3D) case. After playing around with a few approaches this seemed to be the most performant.